lib/Conversion/HWToSV/HWToSV.cpp

```diff
@@ -0,0 +1,83 @@
+//===- HWToSV.cpp - HW To SV Conversion Pass ------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This is the main HW to SV Conversion Pass Implementation.
+//
+//===----------------------------------------------------------------------===//
+
+#include "circt/Conversion/HWToSV.h"
+#include "../PassDetail.h"
+#include "circt/Dialect/HW/HWOps.h"
+#include "circt/Dialect/SV/SVOps.h"
+#include "mlir/Transforms/DialectConversion.h"
+#include "llvm/ADT/TypeSwitch.h"
+
+using namespace mlir;
+using namespace circt;
+using namespace hw;
+using namespace sv;
+
+static sv::EventControl hwToSvEventControl(hw::EventControl ec) {
+  switch (ec) {
+  case hw::EventControl::AtPosEdge:
+    return sv::EventControl::AtPosEdge;
+  case hw::EventControl::AtNegEdge:
+    return sv::EventControl::AtNegEdge;
+  case hw::EventControl::AtEdge:
+    return sv::EventControl::AtEdge;
+  }
+  llvm_unreachable("Unknown event control kind");
+}
```

Comment:
Now that we have `EventControl` in HW, we might be able to replace all other dialect-local `EventControl` things with the HW flavor! I think I've seen a few dialects where we redefine that enum.

Reply:
Captured in https://github.com/llvm/circt/issues/5601.